CMakeLists.txt

```diff
@@ -442,14 +444,9 @@ if(SWIFT_ENABLE_EXPERIMENTAL_CONCURRENCY AND NOT SWIFT_STDLIB_SINGLE_THREADED_RU
 endif()
 
 set(SWIFT_BUILD_HOST_DISPATCH FALSE)
-if(SWIFT_BUILD_SYNTAXPARSERLIB OR SWIFT_BUILD_SOURCEKIT)
-  if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
+if(SWIFT_ENABLE_DISPATCH AND NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
+  if(SWIFT_BUILD_SYNTAXPARSERLIB OR SWIFT_BUILD_SOURCEKIT OR SWIFT_CONCURRENCY_USES_DISPATCH)
```

Comment:
I like the general idea of this pull, but this line subtly changes the logic: now if `SWIFT_CONCURRENCY_USES_DISPATCH` is true, then `SWIFT_BUILD_HOST_DISPATCH` is also set to true. But the two are independent, the former must only be set when the target concurrency libraries are built and the latter when the first two host libraries are built. This won't matter when you're only building everything for the host, but if you're cross-compiling the standalone stdlib, as I do for Android, this will pull in the host dispatch build unnecessarily.